core/txbuilder/finalize.go

```diff
@@ -57,10 +57,13 @@ func FinalizeTx(ctx context.Context, c *protocol.Chain, s Submitter, tx *bc.Tx)
 // least one input to commit to the complete transaction (what you get
 // when you build a transaction with allow_additional_actions=false).
 var ErrNoTxSighashCommitment = errors.New("no commitment to tx sighash")
+var ErrNoTxSighashAttempt = errors.New("no tx sighash attempted")
+var ErrTxSignatureFailure = errors.New("tx signature was attempted but failed")
 
 func checkTxSighashCommitment(tx *bc.Tx) error {
```

Comment:
It may be worth separating the new checks into a function that is called separately, before `checkTxSighashCommitment`. The reasoning being that the two added error checks apply to _all_ inputs, whereas `checkTxSighashCommitment` stops at the first valid input. Perhaps the added signature checks should happen for all inputs.